extensions/wikia/EditPageLayout/SpecialCustomEditPage.class.php

```diff
@@ -471,7 +471,7 @@ public function getWikitextFromRequest() {
 			}
 		}
 
-		return $wikitext;
+		return htmlspecialchars($wikitext);
```

Comment:
So, this at present will break previewing on normal pages as it will escape the wikitext before it is parsed, leading to valid tags in wikitext like `<span></span>` not being handled properly.

Instead of escaping here, move it to the case of CSS and JS pages here:
https://github.com/Wikia/app/blob/479481c93d/extensions/wikia/EditPageLayout/EditPageService.class.php#L83

This would then also match the core handling:
https://github.com/Wikia/app/blob/479481c93d/includes/EditPage.php#L2712